etl/steps/data/garden/moatsos/2023-10-09/moatsos_historical_poverty.py

```diff
@@ -0,0 +1,107 @@
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def create_above_and_between_vars(tb: Table) -> Table:
+    """
+    Create additional variables from the share and number in poverty: above and between poverty lines (or CBN).
+    Also, add metadata to these variables.
+    """
+    # Add population column to estimate other variables
+    tb["pop"] = tb["headcount_cbn"] / (tb["headcount_ratio_cbn"] / 100)
+
+    # Create additional variables
+    # Define columns to use
+
+    poverty_lines = [
+        "190",
+        "500",
+        "1000",
+        "3000",
+    ]
+    cols_number = ["headcount_cbn"]
+    cols_above = []
+    cols_number_above = []
+    cols_between = []
+    cols_number_between = []
+
+    for povline in poverty_lines:
+        tb[f"headcount_{povline}"] = tb[f"headcount_ratio_{povline}"] * tb["pop"] / 100
+        cols_number.append(f"headcount_{povline}")
+
+    # Share of people above poverty lines
+    for povline in poverty_lines:
+        tb[f"headcount_ratio_above_{povline}"] = 100 - tb[f"headcount_ratio_{povline}"]
+        cols_above.append(f"headcount_ratio_above_{povline}")
+        tb[f"headcount_above_{povline}"] = tb[f"headcount_ratio_above_{povline}"] * tb["pop"] / 100
+        cols_number_above.append(f"headcount_above_{povline}")
+
+    # Also do it for cbn
+    tb["headcount_ratio_above_cbn"] = 100 - tb["headcount_ratio_cbn"]
+    cols_above.append("headcount_ratio_above_cbn")
+    tb["headcount_above_cbn"] = tb["headcount_ratio_above_cbn"] * tb["pop"] / 100
+    cols_number_above.append("headcount_above_cbn")
+
+    # Share of people in between poverty lines (World Bank)
+    # For each poverty line in cols_wb
+    for i in range(len(poverty_lines)):
+        if i != 0:
+            tb[f"headcount_ratio_between_{poverty_lines[i-1]}_{poverty_lines[i]}"] = (
+                tb[f"headcount_ratio_{poverty_lines[i]}"] - tb[f"headcount_ratio_{poverty_lines[i-1]}"]
+            )
+            cols_between.append(f"headcount_ratio_between_{poverty_lines[i-1]}_{poverty_lines[i]}")
+
+            tb[f"headcount_between_{poverty_lines[i-1]}_{poverty_lines[i]}"] = (
+                tb[f"headcount_ratio_between_{poverty_lines[i-1]}_{poverty_lines[i]}"] * tb["pop"] / 100
+            )
+            cols_number_between.append(f"headcount_between_{poverty_lines[i-1]}_{poverty_lines[i]}")
+
+    # Round to integer numbers
+    tb[cols_number + cols_number_above + cols_number_between] = tb[
+        cols_number + cols_number_above + cols_number_between
+    ].round()
+
+    tb = geo.harmonize_countries(
+        df=tb,
+        countries_file=paths.country_mapping_path,
+    )
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    # Add metadata by code
+    tb = add_metadata_vars(tb)
```

Comment:
@pabloarosado Hi! I think when adding metadata by code I lose origins. How can I avoid that?

Reply:
There are a few things I don't understand in the code, but I can quickly see that `var_metadata_absolute` (could you please use a function name that describes the purpose of the function?) creates a new variable that does not have origins, so I suppose that's where you are losing them.